exercises/Phil_Colebrooke/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,71 @@
+#include "ShapeSorter.h"
+
+ShapeSorter::ShapeSorter(std::vector<Shape *> shapes) : m_shapes{shapes} {}
+
+/** Prints out the Shapes that match a chosen type
+ *
+ * @param type The Shape type
+ *
+ */
+void ShapeSorter::printByType(std::string type) {
+  for (Shape *shape : m_shapes) {
+    if (shape->getType() == type) {
+      std::cout << shape->getType() + " with perimeter "
+                << shape->getPerimeter() << " and area " << shape->getArea()
+                << "\n";
+    }
+  }
+}
+
+/** Prints out the Shapes that match a chosen number of sides
+ *
+ * @param sides The number of sides
+ *
+ */
+void ShapeSorter::printBySides(int sides) {
+  for (Shape *shape : m_shapes) {
+    if (shape->getSides() == sides) {
+      std::cout << shape->getType() + " with perimeter "
+                << shape->getPerimeter() << " and area " << shape->getArea()
+                << "\n";
```

Comment:
To reduce the number of lines of code and so you don't keep repeating this, you could consider have a print method that is called by each of these methods